src/ethereum/wallet.rs

```diff
@@ -137,6 +165,7 @@ impl Wallet {
         chain_id: ChainId,
     ) -> anyhow::Result<Hash> {
         self.assert_chain(chain_id).await?;
+        let private_key = self.private_key();
```

Comment:
:question: I don't see why this needed changing.

Reply:
It's a function call now.

Reply:
But actually it's an unnecessary change, now fixed.